src/Internal/Wallet/Cip30.js

```diff
@@ -6,15 +6,27 @@ exports._getUtxos = maybe => conn => () =>
   conn.getUtxos().then(res => (res === null ? maybe.nothing : maybe.just(res)));
 
 exports._getCollateral = maybe => conn => () =>
-  conn.experimental
-    .getCollateral()
-    .then(utxos =>
-      utxos !== null && utxos.length ? maybe.just(utxos) : maybe.nothing
-    );
+  /* Notes regarding the quirks of various wallets:
+
+     Yoroi will throw an error if no amount argument is provided, and will
+     break if the following expression is written as
+     (conn.getCollateral || conn.experimental.getCollateral)("5000000") due to
+     JavaScript object binding
+
+     Typhon will throw an error if the amount argument is not a string
+
+     Nami only provides `getCollateral` under the experimental API
+  */
+  (typeof conn.getCollateral === "function"
+    ? conn.getCollateral("5000000")
```

Comment:
This string value is probably incorrect. In the spec, it must be cbor-string of Coin type, not numeric literal.